tests/integration-tests/tests/iam/test_iam.py

```diff
@@ -21,93 +21,192 @@
 from s3_common_utils import check_s3_read_resource, check_s3_read_write_resource, get_policy_resources
 
 from tests.common.assertions import assert_no_errors_in_logs
+from tests.common.schedulers_common import get_scheduler_commands
+from tests.schedulers.test_awsbatch import _test_job_submission as test_job_submission_awsbatch
+from tests.schedulers.test_slurm import _wait_for_computefleet_changed as wait_for_computefleet_changed
 
 
 @pytest.mark.usefixtures("os", "instance")
 def test_iam_roles(
     region,
     scheduler,
-    common_pcluster_policies,
-    role_factory,
-    instance_profile_factory,
+    create_roles_stack,
     pcluster_config_reader,
     clusters_factory,
     test_datadir,
 ):
+
+    cluster_roles_stack = create_roles_stack(
+        stack_prefix="integ-tests-iam-cluster-roles", roles_file="cluster-roles.cfn.yaml"
+    )
+
     is_awsbatch = scheduler == "awsbatch"
     if is_awsbatch:
-        instance_policies = common_pcluster_policies["awsbatch_instance_policy"]
-        lambda_policies = common_pcluster_policies["awsbatch_lambda_policy"]
+        head_instance_role = cluster_roles_stack.cfn_outputs["HeadNodeRoleBatch"]
+        lambda_role = cluster_roles_stack.cfn_outputs["CustomLambdaResourcesRoleBatch"]
+        compute_instance_role = ""
+        compute_instance_profile = ""
     else:
-        instance_policies = common_pcluster_policies["traditional_instance_policy"]
-        lambda_policies = common_pcluster_policies["traditional_lambda_policy"]
-    _, cluster_role_arn = role_factory("ec2", [instance_policies])
-    instance_profile = instance_profile_factory([instance_policies])
-    lambda_role_name, lambda_role_arn = role_factory("lambda", [lambda_policies])
+        head_instance_role = cluster_roles_stack.cfn_outputs["HeadNodeRoleSlurm"]
+        compute_instance_profile = cluster_roles_stack.cfn_outputs["ComputeNodeInstanceProfileSlurm"]
+        compute_instance_role = cluster_roles_stack.cfn_outputs["ComputeNodeRoleSlurm"]
+        lambda_role = cluster_roles_stack.cfn_outputs["CustomLambdaResourcesRoleSlurm"]
```

Comment:
can we move this into a function that returns the tuple, so to avoid also the duplication below?